<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Neil McCracken" />


<title>InflectSSPVignette</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">InflectSSPVignette</h1>
<h4 class="author">Neil McCracken</h4>
<h4 class="date">02-February-2022</h4>



<div id="inflectssp-statistics-string-panther" class="section level1">
<h1>InflectSSP (Statistics, STRING, PANTHER)</h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>InflectSSP is a function that uses LC-MS data from a Thermal Proteome
Profiling (TPP) experiment and user inputs to determine protein melt
shifts across the proteome. There are multiple types of TPP experiments
but the type that can be analyzed using this InflectSSP function is
those where a single type of drug treatment or mutant cell line
(Condition) is used with a single type of vehicle or wild type cell line
(Control). Biological and technical replicate data sets can be analyzed
by this program. These experiments also consist of those where a single
gradient heat treatment (i.e. 25, 35, 40, 50, 55, 60) is used across all
conditions and controls. The output of this program consists of melt
curves for each protein along with summary files that include
bioinformatic analysis. The output files are all saved to the Directory
folder specified by the user and this is the same folder that the user
saves the source input data files. The bioinformatic analysis that is
generated includes overlay of melt shift data with STRING based
annotations.</p>
</div>
<div id="program-inputs" class="section level2">
<h2>Program inputs</h2>
<p>The inputs to the program consist of both MS Excel files along with
user specified parameters that are used by the program to
include/exclude certain data from the melt shift analysis. Inputs
include the following:</p>
<ul>
<li>file inputs
<ul>
<li>MS Excel based files with .xlsx extension</li>
<li>Control data sets (i.e. vehicle) where each file must be labeled as
Control #.xlsx where # is the experiment number. There can be any number
of experiment files for the analysis.</li>
<li>Condition data sets (i.e. treatment) where each file must be labeled
as Condition #.xlsx where # is the experiment number. There can be any
number of experiment files for the analysis.</li>
<li>The columns in each of the files must have the following headers
<ul>
<li>Column 1: Accession</li>
<li>Column 2: PSM</li>
<li>Column 3: UP</li>
<li>Column 4 and higher: The temperatures that were used in each of the
gradients</li>
</ul></li>
<li>The data in each row must correspond to the following:
<ul>
<li>Column 1: The Accession numbers for each protein reported by the
search program. These consist of either a single letter followed by five
digits or 2 letters followed by 6 digits.</li>
<li>Column 2: The number of peptide spectrum matches (PSM) reported for
each protein</li>
<li>Column 3: The number of unique peptides (UP) reported for each
protein</li>
<li>Column 4 and higher: The abundance data for each protein and at each
temperature in the heat treatment</li>
</ul></li>
</ul></li>
<li>function inputs
<ul>
<li>Directory: This consists of the path for the folder where the file
inputs are located on the hard drive of the user. An example format for
this is as follows: “/Users/Einstein/TPPExperiment”</li>
<li>NControl: This is the number of control data sets to be included in
the analysis. Control experiments are those where samples were treated
with vehicle or are the wild type cell line. This number must be at
least 1 and needs to have the same corresponding number of .xlsx files
for every Control experiment.</li>
<li>NCondition: This is the number of Condition data sets to be included
in the analysis. Condition experiments are those where samples were
treated with drug or are the mutated cell line. This number must be at
least 1 and needs to have the same corresponding number of .xlsx files
for every Condition experiment.</li>
<li>PSM: This is the number of Peptide Spectrum Matches that will be
used by the InflectSSP program to determine whether an identified
protein has suitable confidence based on the search algorithm. PSM
values in the Control or Condition data files will need to be greater
than this criteria for the protein to be considered in the final
calculations. The number of PSMs in the raw Control or Condition data
files needs to be listed in the second column of the data set.</li>
<li>UP: This is the number of unique peptides that will be used by the
InflectSSP program to determine whether an identified protein has
suitable confidence based on the search algorithm. UP values in the
Control or Condition data files will need to be greater than this
criteria for the protein to be considered in the final calculations. The
number of UPs in the raw Control or Condition data files needs to be
listed in the third column of the data set.</li>
<li>CurveRsq: This value is the minimum acceptable coefficient of
determination (R squared) for protein melt curves in the experiment.
This number needs to be expressed as a decimal. For example, 0.95
indicates that curves with R squared greater than 0.95 will be
considered in melt shift calculations.</li>
<li>PValMelt: This value is the limit for the p-value of melt shifts
calculated by the program. Melt shifts with p-values greater than or
equal to this value are considered not significant. This value is
specified as a decimal. For instance a value of 0.05 means that proteins
with melt shifts have a p-value of 0.04 would be considered
significant.</li>
<li>MeltLimit: This value is the limit used for the melt shift
magnitude. This limit is used to specify which proteins are considered
significant in the evaluation. Melts greater than this limit are
considered to be significant.</li>
<li>RunSTRING: A Yes or No input that indicates whether or not the
STRING network evaluation is conducted by the program workflow.</li>
<li>STRINGScore: This is the STRING confidence score that is used by the
program to determine which interactions reported by STRING are
significant. This value is expressed as a decimal.</li>
<li>Species: Taxon number for the species. This is used by STRING and
PANTHER. Example is 9606 for human.</li>
</ul></li>
</ul>
</div>
<div id="program-outputs" class="section level2">
<h2>Program outputs</h2>
<p>The program reports the following in a folder titled “Result Files”
in the Directory folder specified by the user:</p>
<ul>
<li>In a folder titled “Good Fit Curves” the melt curves (condition and
control) for each protein that has been reported to have acceptable:
<ul>
<li>PSMs</li>
<li>UPs</li>
<li>Quality of fit for melt curves as measured by R squared</li>
</ul></li>
<li>AllMeltShifts.csv. This file contains all melt shifts that pass PSM
and UP criteria.</li>
<li>AnalysisResults.csv. This file contains data along with calculations
for each protein that was identified as having sufficient quantity of
PSM and UP</li>
<li>FilteredMeltShifts.csv: This file contains the melt shifts that pass
all criteria entered by the user</li>
<li>WaterfallPlot.pdf plot: This is a plot of melt shifts for all
proteins that are numbered in order from lowest melt shift to highest
melt shift. This plot includes proteins that had good quality melts and
met PSM/UP criteria. Red dots are those that are considered to be
significant (melt shift p-value, melt shift magnitude) while black dots
are those that do not meet these criteria.</li>
<li>STRINGNetwork.svg: This file contains and image of a network diagram
that contains nodes for each protein that was observed to be significant
in the analysis pipeline. The lines between each node indicate an
interaction or relationship as identified by STRING according to the
confidence limits specified by the user. The color of each node
corresponds to the melt shift identified by the program.</li>
<li>NodeMelts.csv: This file summarizes the interactions used by the
plotting function to generate the STRING diagram.</li>
<li>Proteins_With_Interactions.csv: This file summarizes the proteins
that have both significant melt shifts and are reported to have
interactions based on STRING.</li>
</ul>
</div>
<div id="general-program-analysis-description" class="section level2">
<h2>General program analysis description</h2>
<p>The following functions are run by the InflectSSP function</p>
<ul>
<li>Import: This function imports the mass spec data from the xlsx
files. The NControl and NCondition specified by the user set the number
of files that are imported.</li>
<li>Normalize: This function normalizes the abundance values for each
protein and at each temperature to the abundance at the lowest
temperature for each protein.</li>
<li>Quantify: This function determines the median abundance at each
temperature across all files in the data sets.</li>
<li>CurveFit1: This function fits the median abundance vs. temperature
using a 4 parameter log fit. In the event that the curve fitting
function is unable to converge, a 3 parameter log fit is used.</li>
<li>Correction: This function corrects the normalized abundance values
for each protein and in each replicate condition or control. The
correction factor is first calculated using the actual and predicted
values from CurveFit1 function. The correction constants are calculated
as follows: (1-((Actual)-Predicted)/((Actual))). The constants are
calculated for each temperature and are multiplied by the normalized
abundance values for each protein in the data sets.</li>
<li>CurveFit2: This function fits 4 parameter log fit curves for each
corrected and normalized abundance value for each protein condition and
control. In the event that either there is no convergence for the fit, a
3 parameter log fit is used. In the event that the melt temperature is
greater than or less than the lowest or highest temperature in the heat
treatment, a 3 parameter log fit is used for the fitting.</li>
<li>MeltCalc: This function calculates the melt temperature for each
protein condition and control using the inflection point in the curve.
The melt shift is calculated using the melt temperatures for each
protein (Condition-Control). The criteria including the p-value for the
melt, melt magnitude and quality of fit for the curves is evaluated and
documented.</li>
<li>ReportDataMelts: This function reports the melt shift related
files</li>
<li>ReportSTRING: This function creates the STRING / melt shift network
diagram</li>
</ul>
</div>
<div id="example-code-for-execution-of-the-function" class="section level2">
<h2>Example code for execution of the function</h2>
<p>Directory&lt;-“/Users/Einstein/TPPExperiment”<br />
NControl&lt;-3<br />
NCondition&lt;-3<br />
PSM&lt;-2<br />
UP&lt;-3<br />
CurveRsq&lt;-0<br />
PValMelt&lt;-0.05<br />
MeltLimit&lt;-0<br />
RunSTRING&lt;-“Yes”<br />
STRINGScore&lt;-0.95<br />
RunPANTHER&lt;-“Yes”<br />
Species&lt;-9606<br />
PANTHERpvalue&lt;-0.05</p>
<p>InflectSSP::InflectSSP(Directory,NControl,NCondition,PSM,UP,CurveRsq,PValMelt,MeltLimit,RunSTRING,STRINGScore,Species)</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(InflectSSP)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Registered S3 method overwritten by &#39;GGally&#39;:</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   method from   </span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   +.gg   ggplot2</span></span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
