<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Neil McCracken" />


<title>InflectSSPVignette</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">InflectSSPVignette</h1>
<h4 class="author">Neil McCracken</h4>
<h4 class="date">01-October-2021</h4>



<div id="inflectssp-statistics-string-panther" class="section level1">
<h1>InflectSSP (Statistics, STRING, PANTHER)</h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>InflectSSP is a function that uses LC-MS data from a Thermal Proteome Profiling (TPP) experiment and user inputs to determine protein melt shifts across the proteome. There are multiple types of TPP experiments but the type that can be analyzed using this InflectSSP function is those where a single type of drug treatment or mutant cell line (Condition) is used with a single type of vehicle or wild type cell line (Control). Biological and technical replicate data sets can be analyzed by this program. These experiments also consist of those where a single gradient heat treatment (i.e. 25, 35, 40, 50, 55, 60) is used across all conditions and controls. The output of this program consists of melt curves for each protein along with summary files that include bioinformatic analysis. The output files are all saved to the Directory folder specified by the user and this is the same folder that the user saves the source input data files. The bioinformatic analysis that is generated includes overlay of melt shift data with STRING and PANTHER based annotations.</p>
</div>
<div id="program-inputs" class="section level2">
<h2>Program inputs</h2>
<p>The inputs to the program consist of both MS Excel files along with user specified parameters that are used by the program to include/exclude certain data from the melt shift analysis. Inputs include the following:</p>
<ul>
<li>file inputs
<ul>
<li>MS Excel based files with .xlsx extension</li>
<li>Control data sets (i.e. vehicle) where each file must be labeled as Control #.xlsx where # is the experiment number. There can be any number of experiment files for the analysis.</li>
<li>Condition data sets (i.e. treatment) where each file must be labeled as Condition #.xlsx where # is the experiment number. There can be any number of experiment files for the analysis.</li>
<li>The columns in each of the files must have the following headers
<ul>
<li>Column 1: Accession</li>
<li>Column 2: PSM</li>
<li>Column 3: UP</li>
<li>Column 4 and higher: The temperatures that were used in each of the gradients</li>
</ul></li>
<li>The data in each row must correspond to the following:
<ul>
<li>Column 1: The Accession numbers for each protein reported by the search program. These consist of either a single letter followed by five digits or 2 letters followed by 6 digits.</li>
<li>Column 2: The number of peptide spectrum matches (PSM) reported for each protein</li>
<li>Column 3: The number of unique peptides (UP) reported for each protein</li>
<li>Column 4 and higher: The abundance data for each protein and at each temperature in the heat treatment</li>
</ul></li>
</ul></li>
<li>function inputs
<ul>
<li>Directory: This consists of the path for the folder where the file inputs are located on the hard drive of the user. An example format for this is as follows: “/Users/Einstein/TPPExperiment”</li>
<li>NControl: This is the number of control data sets to be included in the analysis. Control experiments are those where samples were treated with vehicle or are the wild type cell line. This number must be at least 1 and needs to have the same corresponding number of .xlsx files for every Control experiment.</li>
<li>NCondition: This is the number of Condition data sets to be included in the analysis. Condition experiments are those where samples were treated with drug or are the mutated cell line. This number must be at least 1 and needs to have the same corresponding number of .xlsx files for every Condition experiment.</li>
<li>PSM: This is the number of Peptide Spectrum Matches that will be used by the InflectSSP program to determine whether an identified protein has suitable confidence based on the search algorithm. PSM values in the Control or Condition data files will need to be greater than this criteria for the protein to be considered in the final calculations. The number of PSMs in the raw Control or Condition data files needs to be listed in the second column of the data set.</li>
<li>UP: This is the number of unique peptides that will be used by the InflectSSP program to determine whether an identified protein has suitable confidence based on the search algorithm. UP values in the Control or Condition data files will need to be greater than this criteria for the protein to be considered in the final calculations. The number of UPs in the raw Control or Condition data files needs to be listed in the third column of the data set.</li>
<li>CurveRsq: This value is the minimum acceptable coefficient of determination (R squared) for protein melt curves in the experiment. This number needs to be expressed as a decimal. For example, 0.95 indicates that curves with R squared greater than 0.95 will be considered in melt shift calculations.</li>
<li>PValMelt: This value is the limit for the p-value of melt shifts calculated by the program. Melt shifts with p-values greater than or equal to this value are considered not significant. This value is specified as a decimal. For instance a value of 0.05 means that proteins with melt shifts have a p-value of 0.04 would be considered significant.</li>
<li>MeltLimit: This value is the limit used for the melt shift magnitude. This limit is used to specify which proteins are considered significant in the evaluation. Melts greater than this limit are considered to be significant.</li>
<li>RunSTRING: A Yes or No input that indicates whether or not the STRING network evaluation is conducted by the program workflow.</li>
<li>STRINGScore: This is the STRING confidence score that is used by the program to determine which interactions reported by STRING are significant. This value is expressed as a decimal.</li>
<li>Species: Taxon number for the species. This is used by STRING and PANTHER. Example is 9606 for human.</li>
<li>RunPANTHER: A Yes or No input that indicates whether or not the PANTHER GO term analysis is conducted by the program workflow.</li>
<li>PANTHERpvalue: The decimal that corresponds to the p-value limit specified by PANTHER and which is used by the program to determine which GO terms to report.</li>
</ul></li>
</ul>
</div>
<div id="program-outputs" class="section level2">
<h2>Program outputs</h2>
<p>The program reports the following in a folder titled “Result Files” in the Directory folder specified by the user:</p>
<ul>
<li>In a folder titled “Good Fit Curves” the melt curves (condition and control) for each protein that has been reported to have acceptable:
<ul>
<li>PSMs</li>
<li>UPs</li>
<li>Quality of fit for melt curves as measured by R squared</li>
</ul></li>
<li>AllMeltShifts.csv. This file contains all melt shifts that pass PSM and UP criteria.</li>
<li>AnalysisResults.csv. This file contains data along with calculations for each protein that was identified as having sufficient quantity of PSM and UP</li>
<li>GoodQualityMeltShifts.csv: This file contains the melt shifts that pass all criteria entered by the user</li>
<li>Melt_vs_Abundance.pdf plot: This is a plot of melt shift vs. abundance for proteins that had good quality melts and met PSM/UP criteria. Red dots are those that are considered to be significant (melt shift p-value, melt shift magnitude) while black dots are those that do not meet these criteria.</li>
<li>WaterfallPlot.pdf plot: This is a plot of melt shifts for all proteins that are numbered in order from lowest melt shift to highest melt shift. This plot includes proteins that had good quality melts and met PSM/UP criteria. Red dots are those that are considered to be significant (melt shift p-value, melt shift magnitude) while black dots are those that do not meet these criteria.</li>
<li>PANTHER_BP.svg: This file contains an image of a plot that on the y-axis includes Biological Process GO terms that correspond with the significant proteins in the analysis. On the x-axis are the melt shifts for the proteins. Each dot corresponds to a protein. The color of the dot and the intensity of the blue corresponds with the p-value in PANTHER and the size of the dot corresponds with the p-value for the melt shift. Dots that lie further to the right, are larger in diameter and are a darker shade of blue are more significant.</li>
<li>PANTHER_CC.svg: This file contains an image of a plot that on the y-axis includes Celluclar Component GO terms that correspond with the significant proteins in the analysis. See The BP description above for meaning of data in the plot.</li>
<li>PANTHER_MF.svg: This file contains an image of a plot that on the y-axis includes Molecular Function GO terms that correspond with the significant proteins in the analysis. See The BP description above for meaning of data in the plot.</li>
<li>STRINGNetwork.svg: This file contains and image of a network diagram that contains nodes for each protein that was observed to be significant in the analysis pipeline. The lines between each node indicate an interaction or relationship as identified by STRING according to the confidence limits specified by the user. The color of each node corresponds to the melt shift identified by the program.</li>
</ul>
</div>
<div id="general-program-analysis-description" class="section level2">
<h2>General program analysis description</h2>
<p>The following functions are run by the InflectSSP function</p>
<ul>
<li>Import: This function imports the mass spec data from the xlsx files. The NControl and NCondition specified by the user set the number of files that are imported.</li>
<li>Normalize: This function normalizes the abundance values for each protein and at each temperature to the abundance at the lowest temperature for each protein.</li>
<li>Quantify: This function determines the median abundance at each temperature across all files in the data sets.</li>
<li>CurveFit1: This function fits the median abundance vs. temperature using a 4 parameter log fit. In the event that the curve fitting function is unable to converge, a 3 parameter log fit is used.</li>
<li>Correction: This function corrects the normalized abundance values for each protein and in each replicate condition or control. The correction factor is first calculated using the actual and predicted values from CurveFit1 function. The correction constants are calculated as follows: (1-((Actual)-Predicted)/((Actual))). The constants are calculated for each temperature and are multiplied by the normalized abundance values for each protein in the data sets.</li>
<li>CurveFit2: This function fits 4 parameter log fit curves for each corrected and normalized abundance value for each protein condition and control. In the event that either there is no convergence for the fit, a 3 parameter log fit is used. In the event that the melt temperature is greater than or less than the lowest or highest temperature in the heat treatment, a 3 parameter log fit is used for the fitting.</li>
<li>MeltCalc: This function calculates the melt temperature for each protein condition and control using the inflection point in the curve. The melt shift is calculated using the melt temperatures for each protein (Condition-Control). The criteria including the p-value for the melt, melt magnitude and quality of fit for the curves is evaluated and documented.</li>
<li>ReportDataMelts: This function reports the melt shift related files</li>
<li>ReportSTRING: This function creates the STRING / melt shift network diagram</li>
<li>ReportPANTHER: This function creates the PANTHER / melt shfit network diagram</li>
</ul>
</div>
<div id="example-code-for-execution-of-the-function" class="section level2">
<h2>Example code for execution of the function</h2>
<p>Directory&lt;-“/Users/Einstein/TPPExperiment”<br />
NControl&lt;-3<br />
NCondition&lt;-3<br />
PSM&lt;-2<br />
UP&lt;-3<br />
CurveRsq&lt;-0<br />
PValMelt&lt;-0.05<br />
MeltLimit&lt;-0<br />
RunSTRING&lt;-“Yes”<br />
STRINGScore&lt;-0.95<br />
RunPANTHER&lt;-“Yes”<br />
Species&lt;-9606<br />
PANTHERpvalue&lt;-0.05</p>
<p>InflectSSP::InflectSSP(Directory,NControl,NCondition,PSM,UP,CurveRsq,PValMelt,MeltLimit,RunSTRING,STRINGScore,RunPANTHER,Species,PANTHERpvalue)</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(InflectSSP)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Registered S3 method overwritten by &#39;GGally&#39;:</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   method from   </span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   +.gg   ggplot2</span></span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
